pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -0,0 +1,66 @@
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/genevalogging"
+	"github.com/Azure/ARO-RP/pkg/util/dns"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (mon *Monitor) emitCertificateExpirationStatuses(ctx context.Context) error {
+	// report NotAfter dates for Geneva (always), Ingress, and API (on managed domain) certificates
+	var certs []x509.Certificate
+
+	mdsdCert, err := mon.getCertificate(ctx, operator.SecretName, operator.Namespace, genevalogging.GenevaCertName)
+	if err != nil {
+		return err
+	}
+	certs = append(certs, *mdsdCert[0])
+
+	if dns.IsManagedDomain(mon.oc.Properties.ClusterProfile.Domain) {
+		infraID := mon.oc.Properties.InfraID
+		ingressCertificateName := infraID + "-ingress"
+		apiCertificateName := infraID + "-apiserver"
+
+		ingressCertificate, err := mon.getCertificate(ctx, ingressCertificateName, operator.Namespace, corev1.TLSCertKey)
+		if err != nil {
+			return err
+		}
+
+		apiCertificate, err := mon.getCertificate(ctx, apiCertificateName, operator.Namespace, corev1.TLSCertKey)
+		if err != nil {
+			return err
+		}
+
+		certs = append(certs, *ingressCertificate[0], *apiCertificate[0])
```

Comment:
I think we should either return just one certificate from getCertificate or return the slices as done now (changing the name to get certificate**s**) but always iterate over the returned value as iterating over a nil slice does not panic. I prefer just to return one and not the slice.

***ingressCertificate[0]** and ***apiCertificate[0]** -> I would ensure those are not nil as I am not sure if we are safe against a nil pointer panic here.

Reply:
I changed the underlying certificate parsing from `x509.GetCertificates` to just `x509.GetCertificate` to help with this. It only parses the first certificate in the chain, but that's the only one we care about checking anyway.